src/dataframe/dataframe.jl

```diff
@@ -952,28 +952,15 @@ Base.convert(::Type{DataFrame}, d::AbstractDict) = DataFrame(d)
 ##
 ##############################################################################
 
-function Base.push!(df::DataFrame, associative::AbstractDict{Symbol,Any})
-    i = 1
-    for nm in _names(df)
-        try
-            push!(df[nm], associative[nm])
-        catch
-            #clean up partial row
-            for j in 1:(i - 1)
-                pop!(df[_names(df)[j]])
-            end
-            msg = "Error adding value to column :$nm."
-            throw(ArgumentError(msg))
-        end
-        i += 1
-    end
-end
-
 function Base.push!(df::DataFrame, associative::AbstractDict)
     i = 1
     for nm in _names(df)
         try
-            val = get(() -> associative[string(nm)], associative, nm)
+            val = get(() -> (Base.depwarn("push!(::DataFrame, ::AbstractDict) with "*
```

Comment:
Won't this warning be printed when `associative` doesn't contain a key for one of `df`'s columns? Then it will be confusing to print this before an error.

BTW, you can use the `do` syntax. Also `associative` could be renamed to `d` or `dict`.

Reply:
good catch. will fix